NetKet/Machine/ffnn.hpp

```diff
@@ -0,0 +1,248 @@
+// Copyright 2018 The Simons Foundation, Inc. - All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Eigen/Dense>
+#include <iostream>
+#include <vector>
+#include "Lookup/lookup.hpp"
+#include "Utils/all_utils.hpp"
+#include "layer.hpp"
+
+#ifndef NETKET_FFNN_HPP
+#define NETKET_FFNN_HPP
+
+namespace netket {
+
+template <typename T>
+class FFNN : public AbstractMachine<T> {
+  using VectorType = typename AbstractMachine<T>::VectorType;
+  using MatrixType = typename AbstractMachine<T>::MatrixType;
+
+  std::vector<std::unique_ptr<AbstractLayer<T>>>
+      layers_;  // Pointers to hidden layers
+
+  std::vector<int> layersizes_;
+  int depth_;
+  int nlayer_;
+  int npar_;
+  int nv_;
+
+  int mynode_;
+
+  const Hilbert &hilbert_;
+
+ public:
+  using StateType = typename AbstractMachine<T>::StateType;
+  using LookupType = typename AbstractMachine<T>::LookupType;
+
+  // constructor
+  explicit FFNN(const Hilbert &hilbert, const json &pars)
+      : nv_(hilbert.Size()), hilbert_(hilbert) {
+    from_json(pars);
+  }
+
+  void from_json(const json &pars) override {
+    auto layers_par = pars["Machine"]["Layers"];
```

Comment:
You can use FieldVal, which will provide a better error message if "Layers" is missing.
https://github.com/netket/netket/blob/66c24d47702bc0610e4fc92d2262475fcb03810a/NetKet/Utils/json_helper.hpp#L59